### ISSUE TITLE
Source code highlighting

### ISSUE THREAD

**Issue by warrenseine** [JeroenDeDauw/GitHub#8]
I'm using this extension to display source code on a documentation wiki so it stays up to date.

I've had to hack the extension to re-render the rendered content with a `<source>` tag around it, which will be rendered by another extension I use, Syntax Highlight GeSHi if I remember correctly. This works fine if my case.

I suppose adding a dependency to the project would not be acceptable, so I'm not going to make a pull request. Instead, I'm simply going to post the patch here so it might be useful to somebody someday.

```
diff --git src/GitHub/GitHubParserHook.php src/GitHub/GitHubParserHook.php
index fa21545..7135c90 100644
--- src/GitHub/GitHubParserHook.php
+++ src/GitHub/GitHubParserHook.php
@@ -33,7 +33,7 @@ class GitHubParserHook implements HookHandler {
    public function handle( Parser $parser, ProcessingResult $result ) {
        $this->setFields( $result );

-       return $this->getRenderedContent();
+       return $this->getRenderedContent($parser);
    }

    protected function setFields( ProcessingResult $result ) {
@@ -44,12 +44,16 @@ class GitHubParserHook implements HookHandler {
        $this->branchName = $params['branch']->getValue();
    }

-   protected function getRenderedContent() {
+   protected function getRenderedContent( Parser $parser ) {
        $content = $this->getFileContent();

        if ( $this->isMarkdownFile() ) {
            $content = $this->renderAsMarkdown( $content );
-       }
+       } else {
+           $content = '<source>' . $content . '</source>';
+           $content = $parser->recursiveTagParse( $content, null );
+       }

        return $content;
    }
```

**Comment by JeroenDeDauw** [comment]
I guess you have a very big todo list :)

**Comment by kettner** [comment]
GeSHi functionality would really help! Also, right now the imported text from github is rendered as 'wiki text', so e.g. "#" becomes a '1'. Is there anyway to change that as well?

**Comment by JeroenDeDauw** [comment]
I've created a new ticket to track that enhancement request: https://github.com/JeroenDeDauw/GitHub/issues/20 (it is not currently possible to do this)

**Comment by warrenseine** [comment]
@JeroenDeDauw I do :)

I didn't take the time to look at it and it was always low priority anyway.

I also switched my documentation system to GitHub, dropping MediaWiki in the mean time. So yeah, I'm not going to contribute this addition, sorry :)

You can close or keep it open if you think it still makes sense.

**Comment by JeroenDeDauw** [comment]
@warrenseine thanks for th update. The enhancement itself makes sense, so keeping this open.

**Comment by danyill** [comment]
I was hoping to simply preparse the text from this extension using e.g.:

```
{{#tag:syntaxhighlight|{{#github:Time-Current Grading.sagews|danyill/time-current-grading}}|lang=python}}
```

Unfortunately that didn't work for me. See also [here](http://webmasters.stackexchange.com/questions/68101/create-mediawiki-template-to-format-source-code)

Would this be a better general purpose solution allowing a range of post-processing options?

**Comment by JeroenDeDauw** [comment]
That'd be nice as well, and generally I think it's good to keep programs simple and making them chainable. Not sure this would always work here though. The markdown output can contain HTML such as tables. Does the `syntaxhighlight` tag deal with those correctly? I kinda doubt it.

**Comment by JeroenDeDauw** [comment]
This was implemented in 1.2
